Neos.FluidAdaptor/Classes/ViewHelpers/Form/CheckboxViewHelper.php

```diff
@@ -89,7 +93,22 @@ public function render($checked = null, $multiple = null)
         }
         if (is_array($propertyValue)) {
             if ($checked === null) {
-                $checked = in_array($valueAttribute, $propertyValue, true);
+                if (TypeHandling::isSimpleType(TypeHandling::getTypeForValue(current($propertyValue))) === false || Arrays::containsMultipleTypes($propertyValue)) {
```

Comment:
I think we can just change this to always walk through the array. `Arrays::containsMultipleTypes()` iterates through it anyways.